bindings/compile.ts

```diff
@@ -0,0 +1,194 @@
+import assert from 'assert';
+
+import { isNil } from '../common/helpers';
+import { bindSolcMethod } from './helpers';
+
+export function setup (solJson, core) {
+  return {
+    compileJson: compileJson(solJson),
+    compileJsonCallback: compileJsonCallback(solJson, core),
+    compileJsonMulti: compileJsonMulti(solJson),
+    compileStandard: compileStandard(solJson, core)
+  };
+}
+
+/**********************
+ * COMPILE
+ **********************/
+
+/**
+ * Returns a binding to the solidity compileJSON method.
+ *
+ * @param solJson
+ */
+function compileJson (solJson) {
+  return bindSolcMethod(solJson, 'compileJSON',
+    'string', ['string', 'number'], null);
+}
+
+/**
+ * Returns a binding to the solidity compileJSONMulti method.
+ *
+ * @param solJson
+ */
+function compileJsonMulti (solJson) {
+  return bindSolcMethod(solJson, 'compileJSONMulti',
+    'string', ['string', 'number'], null);
+}
+
+/**
+ * Returns a binding to the solidity compileJSONCallback method.
+ *
+ * @param solJson
+ */
+function compileJsonCallback (solJson, core) {
+  const compileInternal = bindSolcMethod(solJson, 'compileJSONCallback',
+    'string', ['string', 'number', 'number'], null);
+
+  if (isNil(compileInternal)) return compileInternal;
+
+  return function (input, optimize, readCallback) {
+    return runWithCallbacks(solJson, core, readCallback, compileInternal, [input, optimize]);
+  };
+}
+
+/**
+ * Returns a binding to the solidity solidity_compile method with a fallback to
+ * compileStandard.
+ *
+ * @param solJson
+ * @param core
+ */
+function compileStandard (solJson, core) {
```

Comment:
Why do you really need this dependency injection here? I could see it for soljson since it's loaded at runtime but wouldn't it be simpler to just import `core`?

Reply:
Not sure what you mean, core in this situation is used for the runWithCallbacks methods which in turn is used for reset, addFunction, removeFunction, isVersion6, etc.

Reply:
I mean that you could simply import `core` at the top of `compiler.ts` and not have to pass it as an argument. This is a valid design, especially when you need to isolate things in tests but it also makes the code less straightforward so I'm just wondering if it's worth it here. Is there a particular reason you chose that the design here?

Reply:
Ah, I see what you meant. Looking through it again now I realized that the `core` parameter here is not the `core.ts` module as I assumed.

How about naming these parameters/variables `coreBindings`, `compileBindings` and `supportedMethods`? In the PR we can see the type which makes it better but I think that the name should be readable on its own too.

Reply:
yeah sounds good.